src/Database/QueryCompiler.php

```diff
@@ -97,6 +97,17 @@ public function compile(Query $query, ValueBinder $generator)
             $this->_sqlCompiler($sql, $query, $generator),
             $this->{'_' . $type . 'Parts'}
         );
+
+        // Propagate bound parameters from sub-queries if the
+        // placeholders can be found in the SQL statement.
+        if ($query->valueBinder() !== $generator) {
+            foreach ($query->valueBinder()->bindings() as $binding) {
+                $placeholder = ':' . $binding['placeholder'];
```

Comment:
Yeah, I remember now. This was one of those difficult issue that I was postponing.

Still, your code seems like it will match more that it should, should you make it a regex instead?